.travis.yml

```diff
@@ -0,0 +1,37 @@
+language: scala
+
+sudo: false
+
+scala:
+  - "2.12.4"
+  - "2.11.12"
+
+jdk:
+  - oraclejdk8
+
+addons:
+  apt:
+    packages:
+    - graphviz
+
+before_cache:
+  - find $HOME/.ivy2 -name "ivydata-*.properties" -print -delete
+  - find $HOME/.sbt  -name "*.lock"               -print -delete
+
+cache:
+  directories:
+    - $HOME/.ivy2/cache
+    - $HOME/.sbt/boot
+
+script:
+  # need to override as the default is to test
+  - sbt -J-XX:ReservedCodeCacheSize=256m ++$TRAVIS_SCALA_VERSION update mimaReportBinaryIssues
+  # make 'git branch' work again
+  - git branch -f "$TRAVIS_BRANCH" && git checkout "$TRAVIS_BRANCH"
+  # check policies, if on master also upload
+  - if [[ "$TRAVIS_SECURE_ENV_VARS" == "true" ]]; then if [[ "$TRAVIS_BRANCH" == "master" && "$TRAVIS_EVENT_TYPE" == "push" ]]; then sbt 'set credentials += Credentials("whitesource", "whitesourcesoftware.com", "", System.getenv("WHITESOURCE_KEY"))' whitesourceCheckPolicies whitesourceUpdate; else sbt 'set credentials += Credentials("whitesource", "whitesourcesoftware.com", "", System.getenv("WHITESOURCE_KEY"))' whitesourceCheckPolicies; fi ; fi
```

Comment:
do we also run this as part of one of the nightly job, as an extra check "before" releasing?

Reply:
Yes it still is in nightlies

Reply:
so whitesource is in: travis on pr validation, runs in parallel to jenkins tests, can be restarted individually, and then in nightlies